solr/core/src/java/org/apache/solr/handler/admin/SystemInfoHandler.java

```diff
@@ -139,8 +140,19 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
     if (solrCloudMode) {
       rsp.add("zkHost", getCoreContainer(req, core).getZkController().getZkServerAddress());
     }
-    if (cc != null)
-      rsp.add( "solr_home", cc.getSolrHome());
+    if (cc != null) {
+      String solrHome = cc.getSolrHome();
+      rsp.add("solr_home", solrHome);
+
+      Path coreRootDirectory = cc.getCoreRootDirectory();
+      if (coreRootDirectory != null) {
```

Comment:
coreRootDirectory is always non-null.  I don't like null checks that are unnecessary; it may feel "safe" but slightly obscures code clarity and suggests possibilities that are impossibilities.